### DISCUSSION TITLE
How to add moving Text overlay on Video

### DISCUSSION THREAD

**Discussion by ujjally** [Zulko/moviepy#1866] [Q&A]
I want to add **moving Text overlay** from (_Left to Right_) or (_Right to Left_) on top of any **full-length video** by using **moviepy** Python module.

I tried the below code, it worked on short videos like 5 sec, and 10 sec but it's not cover long-length videos like 1/2+ minutes. How to do this for all types of video duration/length like 5/10/30+ minutes.

```
from moviepy.editor import *

video_file = VideoFileClip('video.mp4')

def position(t):
    return (45*t, 0*t)

add_text = TextClip("www.YouTube.com",fontsize=30,color="red").set_position(position).set_duration(video_file.duration)

final = CompositeVideoClip([video_file,add_text])

final.write_videofile("test.mp4")
```

**Comment by keikoro** [comment]
Please report this as a proper issue instead of using GitHub's Discussions feature, thank you.
